File: src/ex009.go
package main

import (
	"fmt"
)

func main() {
	MaxN := 1000

	var c int
	for a := 1; a < MaxN; a++ {
		for b := a + 1; b < MaxN; b++ {
			if a+b >= MaxN {
				break
			}
			c = MaxN - a - b
			if a*a+b*b == c*c {
				fmt.Println(a * b * c)
				break
			}
		}
	}
}

File: src/ex009.scala
object Ex009 {
  def pythagorasThreeSets(maxN: Int, s: Int): List[(Int, Int, Int)] = 
    (
      for (x <- (1 until maxN); y <- (x until s); val z = s - x - y
        if (x * x + y * y == z * z))
        yield (x, y, z)
    ).toList
  def execute(): Int = {
    val (x, y, z) = pythagorasThreeSets(1000, 1000).head
    x * y * z
  }
  def main(args: Array[String]) {
    println(execute())
  }
}

